activerecord/lib/active_record/relation/batches.rb

```diff
@@ -195,16 +198,16 @@ def find_in_batches(start: nil, finish: nil, batch_size: 1000, error_on_ignore:
     #   Person.in_batches.each_record(&:party_all_night!)
     #
     # NOTE: Order can be ascending (:asc) or descending (:desc). It is automatically set to
-    # ascending on the primary key ("id ASC").
-    # This also means that this method only works when the primary key is
+    # ascending on the iteration key ("id ASC").
+    # This also means that this method only works when the iteration key is
     # orderable (e.g. an integer or string).
     #
     # NOTE: By its nature, batch processing is subject to race conditions if
     # other processes are modifying the database.
-    def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore: nil, order: :asc)
+    def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore: nil, order: :asc, by: primary_key)
```

Comment:
Since you want this as an array eventually, maybe just start with one?
```suggestion
    def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore: nil, order: :asc, by: [primary_key])
```

Reply:
Skipping this until I learn a bit more from the community - I don't want to confuse callers since composite keys will not be the norm (your suggestion also applies to the start and finish parameters of this and a few other functions)